tests/core/io/test_logger.h

```diff
@@ -0,0 +1,174 @@
+/**************************************************************************/
+/*  test_logger.h                                                         */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_LOGGER_H
+#define TEST_LOGGER_H
+
+#include "core/io/dir_access.h"
+#include "core/io/logger.h"
+#include "tests/test_macros.h"
+
+// #include "drivers/unix/file_access_unix_pipe.h"
+// Ref<FileAccessUnixPipe> _stdout;
+// _stdout.instantiate();
+// main_pipe->open_existing(pipe_out[0], pipe_in[1], p_blocking);
+
+namespace TestLogger {
+
+const int sleep_duration = 1200000;
```

Comment:
```suggestion
constexpr int sleep_duration = 1200000;
```

Also, can we use a shorter sleep duration? This is 1.2 seconds, which can impact test runtimes significantly since we test 6 rotations in total (8 seconds).

Reply:
I know 1.2 seconds is a lot, but the timestamp format that is added to the log files when are rotated is `vformat("%sT%02d:%02d:%02d", timestamp, dt.hour, dt.minute, dt.second);` which is up to seconds. So in order to get a new file name more than a second must pass between file rotations.